spec/services/send_add_email_confirmation_spec.rb

```diff
@@ -0,0 +1,32 @@
+require 'rails_helper'
+
+RSpec.describe SendAddEmailConfirmation do
+  describe '#call' do
+    let(:user) { create(:user, confirmed_at: nil) }
+    let(:email_address) { user.email_addresses.take }
+    let(:request_id) { '1234-abcd' }
+    let(:confirmation_token) { 'confirm-me' }
+
+    before do
+      allow(Devise).to receive(:friendly_token).once.and_return(confirmation_token)
+      email_address.update!(
+        confirmed_at: nil,
+        confirmation_token: nil,
+        confirmation_sent_at: nil,
+      )
+    end
+
+    subject { described_class.new(user) }
+    it 'sends the user an email with a confirmation link and the request id' do
+      email_address.update!(confirmed_at: Time.zone.now)
+
+      subject.call(email_address:, request_id:, in_select_email_flow: nil)
```

Comment:
A few minor comments about `subject`:

- I'd recommend naming subjects, since there can be multiple and it can help disambiguate ([related guideline](https://github.com/rubocop/rspec-style-guide#use-subject))
- As implemented here, I would not expect a `subject` referring to an instance of the class to be defined within the `describe '#call' do` method tests, but rather outside. I'd expect a `subject` inside the method spec to be the result of that method call.
- I'd expect the `subject` to be defined earlier in the block, alongside the `let` (the earlier-referenced guidelines mention to [have it be the first](https://github.com/rubocop/rspec-style-guide#leading-subject), though I've not always been consistent with this)

```suggestion
  subject(:instance) { described_class.new(user) }

  describe '#call' do
    subject(:result) { instance.call(email_address:, request_id:, in_select_email_flow:) }
    let(:user) { create(:user, confirmed_at: nil) }
    let(:email_address) { user.email_addresses.take }
    let(:request_id) { '1234-abcd' }
    let(:in_select_email_flow) { nil }
    let(:confirmation_token) { 'confirm-me' }

    before do
      allow(Devise).to receive(:friendly_token).once.and_return(confirmation_token)
      email_address.update!(
        confirmed_at: nil,
        confirmation_token: nil,
        confirmation_sent_at: nil,
      )
    end

    it 'sends the user an email with a confirmation link and the request id' do
      email_address.update!(confirmed_at: Time.zone.now)

      result

```

Reply:
Thank you! That style guide is a good read. 🤔